### ISSUE TITLE
Error: 'float round(float)' conflicts with a previous declaration

### ISSUE THREAD

**Issue by sagitter** [assaultcube/AC#147]
Hi all,

AC-code is not compiled with GCC-6.0:

```
+ cd AC-1.2.0.2
+ make -C source/src -j2 CXX=g++ CC=gcc 'CXXFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -fPIC -pie' 'CFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -Wl,-z,relro -fPIC -pie -Wl,-z,now' 'CXXOPTFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -fPIC -pie'
make: Entering directory '/builddir/build/BUILD/AC-1.2.0.2/source/src'
cd ../enet; ./configure --enable-shared=no --enable-static=yes
g++ -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -fPIC -pie -Wall -fsigned-char -I. -Ibot -I../enet/include -I/usr/include `sdl-config --cflags` -idirafter ../include -o cube.h.gch.tmp cube.h
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... In file included from cube.h:8:0:
tools.h: In function 'float round(float)':
tools.h:58:34: error: 'float round(float)' conflicts with a previous declaration
 static inline float round(float x) { return floor(x + 0.5f); }
                                  ^
In file included from /usr/include/c++/6.0.0/math.h:36:0,
                 from platform.h:9,
                 from cube.h:7:
/usr/include/c++/6.0.0/cmath:1708:3: note: previous declaration 'constexpr float std::round(float)'
   round(float __x)
   ^~~~~
In file included from cube.h:69:0:
protos.h: In member function 'bool servercommandline::checkarg(const char*)':
protos.h:1032:48: warning: statement is indented as if it were guarded by... [-Wmisleading-indentation]
                 else if(ai > 0) maxdemos = ai; break;
                                                ^~~~~
protos.h:1032:17: note: ...this 'else' clause, but it is not
                 else if(ai > 0) maxdemos = ai; break;
                 ^~~~
yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... Makefile:132: recipe for target 'cube.h.gch' failed
make: *** [cube.h.gch] Error 1
make: *** Waiting for unfinished jobs....
gcc3
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for gethostbyaddr_r... yes
checking for gethostbyname_r... yes
checking for poll... yes
checking for fcntl... yes
checking for inet_pton... yes
checking for inet_ntop... yes
checking for struct msghdr.msg_flags... yes
checking for socklen_t... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libenet.pc
config.status: executing depfiles commands
config.status: executing libtool commands
make: Leaving directory '/builddir/build/BUILD/AC-1.2.0.2/source/src'
```

**Comment by ac-stef** [comment]
please use branch "next"

**Comment by sagitter** [comment]
Please, see #148.

**Comment by sagitter** [comment]
Fixed on 752950989b4e286459ca9aee3d61a868d7b20fa4
